idmapping-rest/src/test/java/org/uniprot/api/idmapping/queue/IdMappingMessageListenerIT.java

```diff
@@ -0,0 +1,282 @@
+package org.uniprot.api.idmapping.queue;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageBuilder;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.http.MediaType;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.uniprot.api.idmapping.controller.request.IdMappingDownloadRequestImpl;
+import org.uniprot.api.idmapping.model.*;
+import org.uniprot.api.idmapping.service.IdMappingJobCacheService;
+import org.uniprot.api.rest.download.model.DownloadJob;
+import org.uniprot.api.rest.download.model.JobStatus;
+import org.uniprot.api.rest.download.queue.DownloadConfigProperties;
+import org.uniprot.api.rest.download.repository.DownloadJobRepository;
+import org.uniprot.api.rest.output.context.FileType;
+import org.uniprot.api.rest.request.idmapping.IdMappingJobRequest;
+
+@ExtendWith({MockitoExtension.class})
+class IdMappingMessageListenerIT {
+
+    @Mock private MessageConverter converter;
+
+    @Mock DownloadConfigProperties downloadConfigProperties;
+
+    @Mock DownloadJobRepository jobRepository;
+
+    @Mock private RabbitTemplate rabbitTemplate;
+
+    @Mock IdMappingDownloadResultWriterFactory writerFactory;
+
+    @Mock private IdMappingJobCacheService idMappingJobCacheService;
+
+    @Mock UniProtKBIdMappingDownloadResultWriter uniProtKBIdMappingDownloadResultWriter;
+
+    @InjectMocks private IdMappingMessageListener idMappingMessageListener;
+
+    @Test
```

Comment:
Please take out code to create following objects in a creator method:
1. IdMappingDownloadRequestImpl
2. IdMappingJob
3. DownloadJob etc